lrauv_ignition_plugins/src/Interpolation.hh

```diff
@@ -0,0 +1,1231 @@
+    //   << zSliceInds2[interpolatorInds2NewCloud[i]] << std::endl;
+    // igndbg << "newToOldInds[zSliceInds2[interpolatorInds2NewCloud[i]]]: "
+    //   << newToOldInds[zSliceInds2[interpolatorInds2NewCloud[i]]] << std::endl;
+  }
+
+  // if (this->dataPtr->debugGeneral)
+  // {
+  //   igndbg << "FindTrilinearInterpolators(): result indices:" << std::endl;
+  //   for (int i = 0; i < _interpolatorInds1.size(); ++i)
+  //     igndbg << _interpolatorInds1[i] << " " << std::endl;
+  //   for (int i = 0; i < _interpolatorInds2.size(); ++i)
+  //     igndbg << _interpolatorInds2[i] << " " << std::endl;
+  // }
+}
+
+/////////////////////////////////////////////////
+float Interpolation::InterpolateData(
+  const Eigen::Vector3f &_p,
+  const Eigen::MatrixXf &_xyzs,
+  const std::vector<float> &_values)
+{
+  if (this->dataPtr->method == TRILINEAR)
+  {
+    return this->dataPtr->TrilinearInterpolate(_p, _xyzs, _values);
+  }
+  else if (this->dataPtr->method == BARYCENTRIC)
+  {
+    return this->dataPtr->BarycentricInterpolate(_p, _xyzs, _values);
+  }
+  else
+  {
+    ignerr << "method value invalid. "
+      << "Choose a valid interpolation method." << std::endl;
+    return std::numeric_limits<float>::quiet_NaN();
+  }
+}
+
+/////////////////////////////////////////////////
+bool Interpolation::Debug()
+{
+  return this->dataPtr->debugGeneral;
+}
+
+/////////////////////////////////////////////////
+void Interpolation::SetDebug(bool debug)
+{
+  this->dataPtr->debugGeneral = debug;
+}
+
+/////////////////////////////////////////////////
+bool Interpolation::DebugMath()
+{
+  return this->dataPtr->debugMath;
+}
+
+/////////////////////////////////////////////////
+void Interpolation::SetDebugMath(bool debug)
+{
+  this->dataPtr->debugMath = debug;
+}
+
+/////////////////////////////////////////////////
+InterpolationPrivate::InterpolationPrivate(InterpolationMethod _method)
+  : method(_method)
+{
+}
+
+/////////////////////////////////////////////////
+bool InterpolationPrivate::comparePclPoints(
+  const pcl::PointXYZ &a,
+  const pcl::PointXYZ &b)
+{
+  // Comparison between points is arbitrary. This function is only used for
+  // set operations, not for literal sorting.
+  return a.x < b.x && a.y < b.y && a.z < b.z;
+}
+
+/////////////////////////////////////////////////
+void InterpolationPrivate::CreateDepthSlice(
+  float _depth,
+  pcl::PointCloud<pcl::PointXYZ> &_cloud,
+  pcl::PointCloud<pcl::PointXYZ> &_zSlice,
+  std::vector<int> &_zSliceInds,
+  bool _invert)
+{
+  // Separate a z slice, i.e. points with z equal to that of 1st NN
+  // Pass in _invert=true to get indices of removed points
+  pcl::PassThrough<pcl::PointXYZ> passThruFilter =
+    pcl::PassThrough<pcl::PointXYZ>(true);
+  passThruFilter.setInputCloud(_cloud.makeShared());
+  passThruFilter.setNegative(_invert);
+  passThruFilter.setFilterFieldName("z");
+  passThruFilter.setFilterLimits(_depth - 1e-6, _depth + 1e-6);
```

Comment:
Should we pass this as a parameter somewhere. I realize we aren't exposing this just yet, but we probably should make this tolerance configurable. For now I think just adding a `_tol` parameter with a default value of 1e-6 should be enough.

Reply:
Yeah this was part of #156. Now added in 3d7dfb5